Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,212 @@
+are laid out in this section, their usage is precised in following sections.
+
+Client's `congestion_delay`/`security_delay` can be scheduled on three different basis:
+- a block height
+- a local clock
+- an observed feerate fluctuation
+
+Note, these criterias might be observable by an adversary and we recommend to implementations
+to pad any criteria with a random delay to diminish risks of tx-relay jamming.
+
+We don't provide recommendations for the `congestion delay` as its settings depends of user
+liquidity preferences and available local data to predict the feerate. Confirmation of non
+time-sensitive transactions should happen but a user may prefer to delay them in case of
+mempool-congestion.
+
+We provide recommendations for the `security_delay` as any transaction under this fee-bumping
+policy has a time-sensitive confirmation. A delay of 1 block and a fast-confirmation feerate
+as reported by fee-estimator should be preferred.
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transition to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+
+  - if a counterparty's funding signature is received:
+    - MUST finalize the funding transaction with a local `SIGHASH_ALL` signatures
+    - MUST broadcast the funding transaction
+
+A node:
+
+  - if a funding signature has been sent or a funding transaction broadcast and there is no confirmation after 6 blocks:
+    - MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+    - MUST start a fee-bumping timer of length `congestion_delay`
+    - if `congestion_delay` expires:
+      - MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+      - MUST replace the CPFP transaction according to BIP 125
+      - MUST reschedule timer for next bump
+    - if `contract_maturity_bound` is at least 20 blocks ahead from local tip:
+      - MUST spend a funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+      - MUST start a fee-bumping timer of length `security_delay`
+      - MUST disable `congestion_delay` timer
+    - if `security_delay` expires:
+      - MUST spend a funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+      - MUST replace the CPFP transaction according to BIP 125
+      - MUST reschedule timer for next bump
+    - MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+
+  - if the funding transaction confirms:
+    - MUST track spend of the funding output
+    - MUST advances its off-chain state to the execution phase
+    - MUST note its change output for later spend
+
+A node:
+
+  - if the funding input double-spend confirms:
+    - MUST abort the DLC and prevent further interactions with counterparty
```

Comment:
I think this deserves a rationale, as looking back at this I don't remember why this is necessary

Reply:
Left a rationale. Lmk what you think. IMO best practice to clean off-chain state machine early to avoid further miscs/exploitations.